### ISSUE TITLE
DPL-909 [BUG] Detected 500 error when obtaining new barcode

### ISSUE THREAD

**Issue by emrojo** [sanger/baracoda#426]
**Describe the bug**
Received these error messages:

[16/Sep/2023:12:36:04 +0100] "POST /barcodes/WSI/new HTTP/1.1" 500 32 "-" "python-requests/2.26.0" "-"
[16/Sep/2023:12:36:04 +0100] "POST /barcodes/WSI/new HTTP/1.1" 500 32 "-" "python-requests/2.26.0" "-"
[16/Sep/2023:12:36:04 +0100] "POST /barcodes/WSI/new HTTP/1.1" 500 32 "-" "python-requests/2.26.0" "-"

Baracoda returned 500 error in this case. This ticket is to investigate further what was the cause. Possibly it couldn't handle all requests in parallel.

26/02/24 - Issues with GSU retrieving WSI barcodes. Insufficient logging in Baracoda made this very difficult to identify the problem. Issue was resolved as missing json content-type headers but to make this more discoverable in the future Baracoda should log issues like this.

** Acceptance criteria **

- [ ] Set UAT to debug level logging so we can see all issues.
- [ ] Set production to an appropriate level of logging that will enable us to see why 500 errors occur.
  - Potentially include the incoming request contents like headers.
- [ ] Add tests where required to ensure appropriate error handling and logging messages.

**Contact**
Prasanth (GSU)
Naga (GSU)

**RT Ticket Number**
798083

**To Reproduce**
Steps to reproduce the behaviour:
26/02/24 - Sending a post request to barcodes/'prefix'/new without json content-type header causes a 500 (Unsupported MediaType)

**Expected behaviour**
What was expected to happen.

**Screenshots**

**Desktop (please complete the following information):**
* Windows or Mac?
* Browser Chrome, Firefox, Safari or other?
* Browser version (use 'About' to look up)?

**Additional context**

**Comment by KatyTaylor** [comment]
Chatted with Eduardo and Abdullah about this one. 

We think it should be deprioritised because:
- It happens infrequently (2 errored requests vs hundreds of successful requests in the last 2 weeks).
- GSU have a workaround, which is good coding practice to do anyway (check if the response from our server was an error, and try another request if so) - so it shouldn't be impacting them any more.
- We don't know the cause (although suspect it's something to do with the service 'warming up', as the errored requests seem to fall at the start of a batch of requests.
- We haven't been able to reproduce it.
- I suspect it would be quite time-consuming to solve, and the debugging process might impact production (thinking about adding verbose logging to production).

Putting it back into the triage column for now.

**Comment by SujitDey2022** [comment]
@stevieing @TWJW-SANGER closing this bug as not planned as it was updated with the label WONTFIX
